pkg/proc/gdbserial/gdbserver_conn.go

```diff
@@ -872,21 +857,9 @@ func (conn *gdbConn) queryProcessInfo(pid int) (map[string]string, error) {
 
 	pi := make(map[string]string)
 
-	for len(resp) > 0 {
-		semicolon := bytes.Index(resp, []byte{';'})
-		keyval := resp
-		if semicolon >= 0 {
-			keyval = resp[:semicolon]
-			resp = resp[semicolon+1:]
-		}
-
-		colon := bytes.Index(keyval, []byte{':'})
-		if colon < 0 {
-			continue
-		}
-
-		key := string(keyval[:colon])
-		value := string(keyval[colon+1:])
+	csp := colonSemicolonParser{buf: resp}
+	for csp.next() {
+		key, value := string(csp.key), string(csp.value)
```

Comment:
Should we change to `key, value := csp.key, csp.value` according to [the comment](https://github.com/go-delve/delve/pull/3574/files#r1398331609)?

Reply:
It doesn't matter here because this is only executed once per debugging session